config.m4

```diff
@@ -1,29 +1,53 @@
 PHP_ARG_ENABLE(decimal, whether to enable decimal support,
-[  --enable-decimal           Enable decimal support])
+[  --enable-decimal        Enable decimal support])
+
+PHP_ARG_WITH(libmpdec-path, for libmpdec support,
+[  --with-libmpdec-path    libmpdec path], no, no)
 
 if test "$PHP_DECIMAL" != "no"; then
 
-    AC_MSG_CHECKING([for libmpdec files in default path])
-    for i in /usr/$PHP_LIBDIR $(locate -e /usr/**/libmpdec.so); do
-      if test -r $i; then
-        LIBMPDEC_DIR=$(dirname $i)
+    AC_MSG_CHECKING([for libmpdec library in default path])
+    for i in $PHP_LIBMPDEC_PATH /usr /usr/local/; do
+      if test -r $i/$PHP_LIBDIR/libmpdec.$SHLIB_SUFFIX_NAME || test -r $i/$PHP_LIBDIR/libmpdec.a; then
+        LIBMPDEC_DIR=$i/$PHP_LIBDIR
+        AC_MSG_RESULT(found in $LIBMPDEC_DIR)
+        break
+      elif test -r $i/lib/libmpdec.$SHLIB_SUFFIX_NAME || test -r $i/lib/libmpdec.a; then
+        LIBMPDEC_DIR=$i/lib
         AC_MSG_RESULT(found in $LIBMPDEC_DIR)
         break
       fi
     done
 
-    if test -z "$LIBMPDEC_DIR"; then
+    if test -z "$LIBMPDEC_DIR" && test -r /usr/lib/x86_64-linux-gnu/libmpdec.so ; then
```

Comment:
This should be symlinked under /usr/, I don't think we should need this? Not sure though. Looking at how other extensions are doing it seems a lot more simple. Is libmpdec doing something non-standard here?

Reply:
No, lot of other extensions does something similar, see ext/gmp/config.m4

With last commit, this fix the non-x86_64 arch.

Reply:
And again, proper way is pkg-config, but libmpdec don't support it...

Reply:
I'll email the author about it. :+1: